lib/memberManager.ts

```diff
@@ -66,41 +73,50 @@ export class Manager {
     return this.state.size;
   }
 
-  private async extractMember(
+  private async extractMemberQuads(
     member: Term,
     data: RdfStore,
-  ): Promise<Member | undefined> {
-    let quads: Quad[] = [];
-
+  ): Promise<Quad[]> {
+    const logger = log.extend("extract");
     if (this.shapeMap) {
       if (this.shapeMap.size === 1) {
         // Use the only shape available
-        quads = await this.extractor.extract(
+        return await this.extractor.extract(
           data,
           member,
           Array.from(this.shapeMap.values())[0],
         );
       } else if (this.shapeMap.size > 1) {
         // Find what is the proper shape for this member based on its rdf:type
-        const memberType = getObjects(data, member, RDF.terms.type)[0];
-        if (memberType) {
-          const shapeId = this.shapeMap.get(memberType.value);
+        const types = getObjects(data, member, RDF.terms.type);
+        for (let type of types) {
+          const shapeId = this.shapeMap.get(type.value);
           if (shapeId) {
-            quads = await this.extractor.extract(data, member, shapeId);
+            return await this.extractor.extract(data, member, shapeId);
           }
-        } else {
-          // There is no rdf:type defined for this member. Fallback to CBD extraction
-          quads = await this.extractor.extract(data, member);
         }
-      } else {
-        // Do a simple CBD extraction
-        quads = await this.extractor.extract(data, member);
+
+        logger(
+          "%s (%s) is not part of the configured types %s",
+          member.value,
+          types.map((x) => x.value).join(", "),
+          [...this.shapeMap.keys()].join(", "),
+        );
+
+        return [];
       }
-    } else {
-      // Do a simple CBD extraction
-      quads = await this.extractor.extract(data, member);
     }
 
+    // let the extractor do it's thing
+    return await this.extractor.extract(data, member);
+  }
+
+  private async extractMember(
+    member: Term,
+    data: RdfStore,
+  ): Promise<Member | undefined> {
+    const quads: Quad[] = await this.extractMemberQuads(member, data);
```

Comment:
I think the `extractMemberQuads` as well as the `extract` method coming from the cbd shape package can still reject their promise without it being handled.